prboom2/src/dsda/time.c

```diff
@@ -15,58 +15,37 @@
 //	DSDA Time
 //
 
+#ifdef _WIN32
+#include <windows.h>
+#else
 #include <time.h>
+#endif
 
 #include "i_system.h"
 
 #include "time.h"
 
-// [XA] MSVC hack: need to implement a handful of functions here.
-// this was yoinked from https://stackoverflow.com/a/31335254 so
-// no guarantees on if this stuff is accurate/kosher/whatever,
-// so maybe don't cut an official release with MSVC just yet.
-// it's enough to hack it into compiling for me, at least. ;)
-
-#ifdef _MSC_VER
-
-#include <windows.h>
+// lovey: Win32 interface
+#ifdef _WIN32
 
-#define CLOCK_MONOTONIC -1
-#define exp7           10000000i64 //1E+7
-#define exp9         1000000000i64 //1E+9
-#define w2ux 116444736000000000i64 //1.jan1601 to 1.jan1970
+static unsigned long long dsda_time[DSDA_TIMER_COUNT];
 
-void unix_time(struct timespec *spec) {
-  __int64 wintime; GetSystemTimeAsFileTime((FILETIME*)&wintime);
-  wintime -= w2ux;  spec->tv_sec = wintime / exp7;
-  spec->tv_nsec = wintime % exp7 * 100;
+void dsda_StartTimer(int timer) {
+  QueryPerformanceCounter((LARGE_INTEGER*)&dsda_time[timer]);
 }
 
-int clock_gettime(int _skip, struct timespec *spec)
-{
-  static struct timespec startspec;
-  static double ticks2nano;
-  static __int64 startticks, tps = 0;
-  __int64 tmp, curticks;
-
-  QueryPerformanceFrequency((LARGE_INTEGER*)&tmp);
+unsigned long long dsda_ElapsedTime(int timer) {
+  static unsigned long long freq = 0;
+  unsigned long long now;
 
-  if (tps != tmp) {
-    tps = tmp;
-    QueryPerformanceCounter((LARGE_INTEGER*)&startticks);
-    unix_time(&startspec); ticks2nano = (double)exp9 / tps;
-  }
+  if (!freq) QueryPerformanceFrequency((LARGE_INTEGER*)&freq);
```

Comment:
Please follow the style elsewhere that adds spaces in between operators, casts, etc. (I realize the code you are replacing wasn't matching either.)

Reply:
Sorry, I wasn't really thinking of code formatting when I made this, but now that I think about it it's a big deal.

I'll take this as a learning experience